addons/pos_cache/models/pos_cache.py

```diff
@@ -23,10 +23,10 @@ def refresh_all_caches(self):
 
     @api.one
     def refresh_cache(self):
-        products = self.env['product.product'].search(self.get_product_domain())
+        products = self.env['product.product'].sudo(
+            self.compute_user_id.id).search(self.get_product_domain())
```

Comment:
The code is more readable like this:

products = self.env['product.product'].sudo(self.compute_user_id.id)\
    .search(self.get_product_domain())

Reply:
Even better as:
```python
Product = self.env['product.product'].sudo(self.compute_user_id)
products = Product.search(self.get_product_domain())
```